Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -122,15 +135,123 @@ public class OptimizelyUserContext {
         guard let optimizely = self.optimizely, optimizely.config != nil else {
             throw OptimizelyError.sdkNotReady
         }
-
+        
         try optimizely.track(eventKey: eventKey,
                              userId: userId,
                              attributes: attributes,
                              eventTags: eventTags)
     }
+    
+}
+
+// MARK: - ForcedDecisions
 
+extension OptimizelyUserContext {
+    
+    /// Sets the forced decision (variation key) for a given flag and an optional rule.
+    /// - Parameters:
+    ///   - flagKey: A flag key.
+    ///   - ruleKey: An experiment or delivery rule key (optional).
+    ///   - variationKey: A variation key.
+    /// - Returns: true if the forced decision has been set successfully.
+    public func setForcedDecision(flagKey: String,
```

Comment:
what if empty flagKey is passed here?

Reply:
It'll be considered as just like other invalid (no-matching) flagKeys. Instead of filtering the invalid keys here, we validate the forced-decision mapping when used for decision.